core/network/src/p2p/gossip/gossip.rs

```diff
@@ -0,0 +1,321 @@
+use crate::p2p::client::{AppResponseCallback, Client};
+use crate::p2p::gossip::{Gossipable, Set};
+use crate::p2p::sdk::{PullGossipRequest, PullGossipResponse};
+use avalanche_types::ids::Id;
+use log::{debug, error};
+use prost::Message;
+use std::error::Error;
+use std::hash::Hash;
+use std::ops::Deref;
+use tokio::sync::Mutex;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::mpsc::{channel, Receiver};
+use tokio::time::interval;
+
+pub struct Config {
+    pub namespace: String,
+    pub frequency: Duration,
+    pub poll_size: usize,
+}
+
+pub struct Gossiper<S: Set + 'static> {
+    config: Config,
+    set: Arc<Mutex<S>>,
+    client: Arc<Mutex<dyn Client>>,
+    stop_rx: Receiver<()>,
+}
+
+unsafe impl<S: Set> Sync for Gossiper<S> {}
+
+unsafe impl<S: Set> Send for Gossiper<S> {}
+
+impl<S> Gossiper<S>
+    where
+        S: Set,
+        S::Item: Default
+{
+    pub fn new(
+        config: Config,
+        set: Arc<Mutex<S>>, // Mutex or RWLock here ?
+        client: Arc<Mutex<dyn Client>>,
+        stop_rx: Receiver<()>,
+    ) -> Self {
+        Self {
+            config,
+            set,
+            client,
+            stop_rx,
+        }
+    }
+
+    pub async fn gossip(&mut self) {
+        let mut gossip_ticker = interval(self.config.frequency);
+
+        loop {
+            select! {
+                _ = gossip_ticker.tick() => {
+                    if let Err(e) = self.execute().await {
+                        error!("Failed to Gossip : {:?}", e);
+                        //ToDo
```

Comment:
Use `todo!();` so you can check to see if the rest of your code compiles properly! I know it doesn't actually make a difference here, but it can be super helpful, you don't have to return proper types.